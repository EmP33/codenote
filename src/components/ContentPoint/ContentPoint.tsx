import React from "react";
import { Typography } from "@mui/material";
import styled from "styled-components";

const ContentDiv = styled.div`
  display: flex;
  justify-content: flex-end;
  z-index: 5;
`;

interface ContentPointProps {
  index: string;
  heading: string;
}

const ContentPoint: React.FC<ContentPointProps> = ({ index, heading }) => {
  return (
    <ContentDiv>
      <Typography
        variant="h6"
        sx={{
          borderTop: "5px solid #fff",
          padding: { xs: "1rem .5rem 0 0", md: "1rem 0 0 0" },
          width: "300px",
          display: "flex",
          justifyContent: "flex-end",
        }}
      >
        <span style={{ color: "var(--color-base)", paddingRight: 3 }}>
          {index}
        </span>
        / {heading}
      </Typography>
    </ContentDiv>
  );
};

export default ContentPoint;
