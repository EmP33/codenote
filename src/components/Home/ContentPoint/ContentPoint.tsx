import React from "react";
import { Typography } from "@mui/material";
import styled from "styled-components";

const ContentDiv = styled.div`
  display: flex;
  justify-content: flex-end;
  z-index: 5;
`;

interface ContentPointProps {
  index: string;
  heading: string;
}

const ContentPoint: React.FC<ContentPointProps> = ({ index, heading }) => {
  return (
    <ContentDiv>
      <Typography
        variant="h6"
        id="heading"
        sx={{
          borderTop: "5px solid var(--color-secondary-dark)",
          padding: { xs: "1rem .5rem 0 0", md: "1rem 0 0 0" },
          width: "300px",
          display: "flex",
          justifyContent: "flex-end",
          color: "var(--color-secondary)",
        }}
      >
        <span style={{ color: "var(--color-secondary-dark)", paddingRight: 3 }}>
          {index}
        </span>
        / {heading}
      </Typography>
    </ContentDiv>
  );
};

export default ContentPoint;
