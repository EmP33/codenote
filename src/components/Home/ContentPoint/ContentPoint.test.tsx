import { shallow } from "enzyme";
import ContentPoint from "./ContentPoint";

describe("ContentPoint component", () => {
  const wrapper = shallow(
    <ContentPoint index="01" heading="Example heading" />
  );
  it("expect to render ContentPoint component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render index and heading properly in heading Typography", () => {
    expect(wrapper.find("#heading").text()).toBe("01/ Example heading");
  });
});
