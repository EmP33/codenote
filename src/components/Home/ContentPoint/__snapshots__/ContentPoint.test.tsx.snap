// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPoint component expect to render ContentPoint component 1`] = `
<styled.div>
  <ForwardRef(Typography)
    id="heading"
    sx={
      Object {
        "borderTop": "5px solid var(--color-secondary-dark)",
        "color": "var(--color-secondary)",
        "display": "flex",
        "justifyContent": "flex-end",
        "padding": Object {
          "md": "1rem 0 0 0",
          "xs": "1rem .5rem 0 0",
        },
        "width": "300px",
      }
    }
    variant="h6"
  >
    <span
      style={
        Object {
          "color": "var(--color-secondary-dark)",
          "paddingRight": 3,
        }
      }
    >
      01
    </span>
    / 
    Example heading
  </ForwardRef(Typography)>
</styled.div>
`;
